feat: add CKB Features in tech explaination (#706)
nervosnetwork/docs.nervos.org: website/docs/tech-explanation/extreme-decentralization.md

---
id: extreme-decentralization

> title: Extreme Decentralization

Reviewer: The title “Extreme Decentralization” may sound negative. Consider alternatives like Uncompromised Decentralization or Maximum Decentralization.

---

> Decentralization is the core value proposition of public blockchains. It ensures censorship resistance, permissionless access, and security without relying on trusted intermediaries. CKB (The layer 1 protocol) is designed from the ground up to make sure this core value is not compromised.

Reviewer: ```suggestion
Decentralization is the core value proposition of public blockchains. It ensures censorship resistance, permissionless access, and security without relying on trusted intermediaries. As the layer 1 protocol of Nervos Network, CKB is designed from the ground up to make sure this core value is not compromised.
```

## Decoupling: Our Approach to True Decentralization

> To achieve true decentralization, we must abandon the view of a blockchain as a monolithic, "do-it-all" platform. If a single blockchain layer tries to be everything to everyone—fast for users, easy for developers, and secure for the network—it inevitably creates conflicting incentives. High throughput demands often override verification needs, and the push for seamless developer experiences frequently introduces centralized dependencies.

Reviewer: ```suggestion
Achieve true decentralization requires moving away from the view of a blockchain as a monolithic, "do-it-all" platform. When a single layer tries to serve everyone—fast for users, easy for developers, and secure for the network—it inevitably faces conflicting incentives. High throughput demands override verification needs, while the push for seamless developer experiences frequently introduces centralized dependencies.
```

> To achieve true decentralization, we must abandon the view of a blockchain as a monolithic, "do-it-all" platform. If a single blockchain layer tries to be everything to everyone—fast for users, easy for developers, and secure for the network—it inevitably creates conflicting incentives. High throughput demands often override verification needs, and the push for seamless developer experiences frequently introduces centralized dependencies.

Reviewer 2: ```suggestion
To achieve true decentralization, a blockchain cannot function as a "do everything" platform. If a single layer attempts to maximize performance, offer developer convenience, and maintain strong security at the same time, these goals create conflicting incentives. High throughput begins to override verification needs, and shortcuts taken for developer experience frequently introduce points of centralization.
```

Nervos CKB addresses this by decoupling concerns across a multi-layered architecture, viewing the blockchain landscape as a continuous spectrum:

- **The Far Left:** Extreme decentralization, lowest-threshold verification, and uncompromised security.
- **The Far Right:** Developer experience, high throughput, the middleware and centralized services needed for mass adoption.

### The Last Line of Defense

> The CKB team consciously positions the Layer 1 protocol at the **absolute left-most edge** of this spectrum. By holding this ground, Layer 1 acts as the immutable trust anchor and the **last line of defense** for the entire network. If the CKB team were to compromise here for the sake of usability or speed, the security of the entire structure would crumble.

Reviewer: ```suggestion
The CKB team consciously positions the Layer 1 protocol at the **far left-most edge** of this spectrum. By holding this ground, Layer 1 acts as the immutable trust anchor and the **last line of defense** for the entire network. Any compromise here for the sake of usability or speed would undermine the security of the entire network.
```

> This philosophy dictates a clear division of labor, ensuring every point on the spectrum receives dedicated effort from the actors best suited for it:

Reviewer 2: ```suggestion
CKB is positioned at the **far-left edge** of this spectrum, serving as the immutable trust anchor and the **last line of defense** for the entire network. Compromising the base layer for usability or speed would weaken the security of the entire structure.

This separation establishes a clear division of labor:
```

It is not only about code, but also for operational decisions. For example, the CKB team might deliberately reject providing centralized conveniences, such as a default, production RPC service (similar to Infura in Ethereum). While such a service would offer a shortcut for developers, it introduces a central point of failure. Note that the [public RPC service](/docs/getting-started/rpcs) that CKB team provide is only for testing purpose, it is strongly recommended to run your own RPC service for your dApp. By refusing to encroach on the "service layer," CKB forces the ecosystem to build its own robust, distributed access layers, ensuring the foundation remains a neutral, decentralized settlement platform.

## How to Achieve Extreme Decentralization

Decentralization is not just about who _mines_ the next block; it is about who can _verify_ the history. If running a node becomes too expensive due to data bloat, the network centralizes into a few server farms regardless of its consensus algorithm.

> CKB ensures this never happens through three radical approaches: **Sustainable State Management**, **Proof-of-Work Consensus**, and a viable **Light Client** with an engineering effort on building nodes that lowers the entry barrier for participants.

Reviewer: ```suggestion
CKB avoid this outcome through three core design choices: **Sustainable State Management**, **Proof-of-Work Consensus**, and a viable **Light Client** with an ongoing engineering effort to lower the barrier of running nodes.
```

> CKB ensures this never happens through three radical approaches: **Sustainable State Management**, **Proof-of-Work Consensus**, and a viable **Light Client** with an engineering effort on building nodes that lowers the entry barrier for participants.

Reviewer 2: ```suggestion
Decentralization is not just about who mines the next block–it is about who can independently **verify** the chain's history. If running a node becomes too expensive due to data bloat, participation collapses into a few data centers, and the system recentralizes in practice regardless of its consensus algorithm.

CKB addresses this challenge through three radical design choices: **Sustainable State Management**, **Proof-of-Work Consensus**, and a viable **Light Client**, supported by ongoing engineering efforts to lower the barrier to running nodes.
```

### 1. Sustainable State Management

Most blockchains operate as "General Computation Networks" (like a world computer). Users pay a one-time fee to execute a transaction, but the resulting data (state) occupies the network's storage forever at no further cost. This leads to the **"Tragedy of the Commons"**: because storage is effectively free, the state grows indefinitely ([State Explosion](https://medium.com/nervosnetwork/state-explosion-and-the-tragedy-of-the-blockchain-commons-1fbd4837e859)). Eventually, only enterprise-grade hardware can store the chain, forcing regular users to trust third parties.

> CKB adopts a "General Verification Network" architecture based on [Cell Model](/docs/tech-explanation/cell-model). The **Common Knowledge Base** acts as a verified state layer rather than a computation layer. To maintain "Extreme Decentralization," CKB creates a direct economic link between state and scarcity:

Reviewer: ```suggestion
CKB, instead, adopts a "General Verification Network" architecture based on the [Cell Model](/docs/tech-explanation/cell-model). The **Common Knowledge Base** acts as a state verification layer rather than a computation layer. To maintain decentralization, CKB directly ties state scarcity with economic cost:
```

- **State as a Scarce Resource:** In CKB, state is not an afterthought; it is the physical "land" of the blockchain.
- **The CKB = Byte Equation:** The native token (CKB) represents **state capacity**. Holding 1 CKB entitles you to store 1 byte of data on the blockchain.

> - **Targeted Inflation (State Rent):** To occupy space on the Global State, users must lock CKB tokens. This incurs an opportunity cost (via secondary issuance inflation), effectively acting as a "State Rent." This forces developers and users to be efficient—they only store what is truly valuable.

Reviewer 2: ```suggestion
CKB takes a different approach. It adopts a "General Verification Network" architecture built on the [Cell Model](/docs/tech-explanation/cell-model), where on-chain storage is treated as a scarce resource rather than free space.

- **State as a Scarce Resource:** On-chain state is modeled as a finite resource that must be explicitly accounted for.
- **1 CKB = 1 Byte of Storage:** The native token, CKB, represents state capacity; holding 1 CKB entitles the holder to store 1 byte of data on the blockchain.
- **Targeted Inflation (State Rent):** Occupying on-chain storage requires locking CKB, which carries an opportunity cost through secondary issuance. This mechanism functions as a form of "state rent" and encourages efficient use of capacity, ensuring that developers and users store only the truly valuable data.
```

- **Merkle Mountain Ranges (MMR):** CKB blocks include an MMR root in their headers. This cryptographic structure allows the light client to prove that any specific block is part of the valid chain history without storing the history itself.
- **Result:** A user can sync the chain in seconds with minimal bandwidth, regardless of how long the blockchain becomes.

#### WASM & In-Browser Verification

The most radical feature of the CKB Light Client is its portability. The CKB team has compiled the light client into **WebAssembly (WASM)**, enabling it to run in environments previously thought impossible for blockchain nodes.

- **No Installation Required:** The light client can be embedded directly into a web wallet or dApp. When a user visits a website, the browser _becomes_ a node.
- **Trustless Interaction:** The dApp does not ask a server "What is my balance?" or "Did this trade settle?" Instead, the browser connects directly to the P2P network, samples headers, and mathematically verifies the data locally.

> - **Mobile Ready:** Because it requires minimal storage (storing only a single block header between executions) and low CPU usage, it runs efficiently on mobile devices without draining battery.

Reviewer 2: ```suggestion
The CKB Light Client is compiled to **WebAssembly (WASM)**, enabling it to run in browser environments and other platforms that do not support traditional full node operations.

- **No Installation Required:** The light client can be embedded directly into a web wallet or dApp. When a user visits a website, the browser operates as a lightweight verification node.
- **Trustless Interaction:** Instead of relying on a server to check balances or transactions, the browser connects to the P2P network, samples block headers, and verifies data locally. 
- **Mobile Ready:** With minimal storage requirements(storing only a single block header between executions) and low CPU usage, the light client can operate efficiently on mobile devices.
```

#### Privacy and Sovereignty

The CKB Light Client protocol allows the client to request specific transaction data and state (Cells) related _only_ to the user.

- **Data Minimization:** The client filters data at the network level. If you own 5 UTXOs (Cells), your device only fetches and verifies the proofs for those 5 items, ignoring the rest of the global state.
- **Censorship Resistance:** Because the client connects to a randomized mesh of peers rather than a single RPC endpoint, it is extremely difficult for any single entity to block the user's access to the network.

By embedding the verification layer directly into the user's application, CKB closes the final gap in decentralization: **The user is not just a customer of the network; the user _is_ the network.**

## Summary

> This completes the picture of "Extreme Decentralization" on CKB: a system where users own their state(Cell Model), participate permissionlessly (PoW), compute locally (Verification Network), and verify independently (Light Client).

Reviewer 2: ```suggestion
In summary, CKB achieves extreme decentralization through three core components:
- **Sustainable State Management**: By enforcing the rule that **1 CKB = 1 byte of storage**, CKB treats state as a scarce resource, preventing unbounded growth and keeping full nodes practical to run.
- **Proof-of-Work Consensus**: Security is anchored in permissionless participation, allowing anyone to help secure the network by contributing computing power.
- **Light Client**: Independent verification is possible on browsers and mobile devices without requiring full-node resources. 
```
